packages/web-components/@sfx/search-box/test/unit/search-box.test.ts

```diff
@@ -0,0 +1,208 @@
+      });
+    });
+
+    describe('value property', () => {
+      it('should have default value of an empty string', () => {
+        expect(searchbox.value).to.equal('');
+      });
+    });
+
+    describe('searchButton property', () => {
+      it('should have default value of false', () => {
+        expect(searchbox.searchButton).to.equal(false);
+      });
+    });
+
+    describe('clearButton property', () => {
+      it('should have default value of false', () => {
+        expect(searchbox.clearButton).to.equal(false);
+      });
+    });
+  });
+
+  describe('connectCallback', () => {
+    it('should add an autocomplete_hover eventListener to the window', () => {
+      const windowAddEventListener = stub(window, 'addEventListener');
+
+      searchbox.connectedCallback();
+
+      expect(windowAddEventListener).to.have.been.calledWith(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, searchbox.updateText);
+    });
+  });
+
+  describe('disconnectCallback', () => {
+    it('should remove the autocomplete_hover eventListener from the window', () => {
+      const windowRemoveEventListener = stub(window, 'removeEventListener');
+
+      searchbox.disconnectedCallback();
+
+      expect(windowRemoveEventListener).to.have.been.calledWith(
+        SEARCHBOX_EVENT.UPDATE_SEARCH_TERM,
+        searchbox.updateText
+      );
+    });
+  });
+
+  describe('emitSearchEvent', () => {
+    it('should dispatch a search request event', () => {
+      const searchRequestEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCH_REQUEST, {
+        detail: 'a',
+        bubbles: true
+      });
+      searchbox.value = 'a';
+
+      searchbox.emitSearchEvent();
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchRequestEvent);
+    });
+  });
+
+  describe('emitSearchBoxClearedEvent', () => {
+    it('should dispatch an emitSearchBoxClearedEvent', () => {
+      const searchboxClearedEvent = new CustomEvent('sfx::search_box_cleared');
+
+      searchbox.emitSearchBoxClearedEvent();
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchboxClearedEvent);
+    });
+  });
+
+  describe('updateText', () => {
+    it('should update the value property in response to data received', () => {
+      const detail = 'inputText';
+      stub(searchbox, 'getInputElement').returns(
+        html`
+          <input type="text" data-sfx-ref="searchInput" placeholder="Type your search" />
+        `
+      );
+      searchbox.updateText({ detail });
+      expect(searchbox.value).to.equal(detail);
+    });
+  });
+
+  describe('handleKeydown', () => {
+    it('should invoke the emitSearchEvent function if enter is pressed and value property length is greater than 0', () => {
+      const emitSearchStub = stub(searchbox, 'emitSearchEvent');
+      searchbox.value = 'hello';
+
+      searchbox.handleKeydown({ keyCode: KEY_CODES.ENTER });
+
+      expect(emitSearchStub).to.have.been.called;
+    });
+  });
```

Comment:
I think you'd want to test the inverses as well: when a key other than "Enter" is pressed, and when the search term is empty.